### PR TITLE
feat(control_data_collecting_tool): add safety measures and rosbag recording and loading

### PR DESCRIPTION
**Pull request by YoshihiroKogure**
## Description

Add following two functionalities in this PR.
- Add a feature to stop when crossing the green line.

[pr_safety_measure.webm](https://github.com/user-attachments/assets/f062b68f-eab7-48e8-8e34-03d00111572d)

- Rosbag2 recording and loading functionalities
  - When nodes are launched, topics in `config/topics.yaml` are automatically recorded and the data is saved in the directory where nodes are launched.
  - The data from `/localization/kinematic_state` and `/localization/acceleration` located in the current directory. You can switch-off this functionality if `LOAD_ROSBAG2_FILES` in `config/param.yaml` is set to `false`.
  Following demo shows control data collecting tool loads data in the current directory and automatically reflects data counting plots. 

[pr_load_data.webm](https://github.com/user-attachments/assets/cffbad87-c10f-4de0-909f-8ecfbb011c2e)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control_data_collecting_tool/scripts/rosbag_play.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        target_topic_included = True
+        if topic_name not in list(topic_types.keys()):
+            target_topic_included = False
+        conn.close()
+
+        if not target_topic_included:
+            return target_topic_included
+
+        #
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/rosbag_play.py:70]:
```suggestion
```

**Comment by YoshihiroKogure** [comment]
In the [00b3e0b](https://github.com/autowarefoundation/autoware_tools/pull/134/commits/00b3e0bfb0881b54b274bbcaf017d564546747c1) commit, I modified it to count data only when the operation mode is set to 3.

File: control_data_collecting_tool/scripts/rosbag_play.py (hunk 1 above)
Comment (YoshihiroKogure) [reply control_data_collecting_tool/scripts/rosbag_play.py]: Thank you for your suggestions. I fixed in [24d4175](https://github.com/autowarefoundation/autoware_tools/pull/134/commits/24d41751bc9af5da762bfb9b50c4660f4f2b1a39).

File: control_data_collecting_tool/scripts/data_collecting_rosbag_record.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        self.timer_callback = self.create_timer(
+            self.timer_period_callback,
+            self.record_message,
+        )
+
+    def subscribe_operation_mode(self, msg):
+        self.present_operation_mode_ = msg.mode
+
+    def record_message(self):
+        # Start subscribing to topics and recording if the operation mode is 3
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/data_collecting_rosbag_record.py:150]:
```suggestion
        # Start subscribing to topics and recording if the operation mode is 3(LOCAL)
```

**Comment by kosuke55** [comment]
I confirmed that I can read the collection status from rosbag

image

File: control_data_collecting_tool/scripts/data_collecting_base_node.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        self.a_bin_centers = (self.a_bins[:-1] + self.a_bins[1:]) / 2
+
+    def onOdometry(self, msg):
+        self._present_kinematic_state = msg
+
+    def onAcceleration(self, msg):
+        self._present_acceleration = msg
+
+    def subscribe_operation_mode(self, msg):
+        self.present_peration_mode_ = msg.mode
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/data_collecting_base_node.py:159]:
```suggestion
        self.present_operation_mode_ = msg.mode
```

Comment (kosuke55) [reply control_data_collecting_tool/scripts/data_collecting_base_node.py:159]: could you please some typos?

File: control_data_collecting_tool/scripts/data_collecting_rosbag_record.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        # Start recording if topics are subscribed and the operation mode is 3
+        if self.present_operation_mode_ == 3 and self.subscribed and not self.recording:
+            self.writer.start_record()
+            self.recording = True
+
+        # Stop recording if the operation mode changes from 3
+        if self.present_operation_mode_ != 3 and self.subscribed and self.recording:
+            self.writer.stop_record()
+            self.subscribed = False
+            self.recording = False
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/data_collecting_rosbag_record.py:165]: please add message print in the terminal when start/stop record

File: control_data_collecting_tool/scripts/data_collecting_base_node.py (hunk 3 above)
Comment (YoshihiroKogure) [reply control_data_collecting_tool/scripts/data_collecting_base_node.py]: Thank you for your feedback. I have made the changes in [c95cdfb](https://github.com/autowarefoundation/autoware_tools/pull/134/commits/c95cdfb26f5ff18858b7fb1a74dc769845b28b98).

File: control_data_collecting_tool/scripts/data_collecting_rosbag_record.py (hunk 4 above)
Comment (YoshihiroKogure) [reply control_data_collecting_tool/scripts/data_collecting_rosbag_record.py:168]: Thank you for your feedback. I have made the changes in [c95cdfb](https://github.com/autowarefoundation/autoware_tools/pull/134/commits/c95cdfb26f5ff18858b7fb1a74dc769845b28b98).

File: control_data_collecting_tool/scripts/data_collecting_base_node.py (hunk 3 above)
Comment (kosuke55) [reply control_data_collecting_tool/scripts/data_collecting_base_node.py]: I rebased on main so please force push if you fix

Comment (YoshihiroKogure) [reply control_data_collecting_tool/scripts/data_collecting_base_node.py]: Thank you for repeatedly pointing out my mistakes. I have made the corrections.

**Review by kosuke55 — APPROVED** [review]
LGTM!

**Pull request merged**